### PR TITLE
[TextAnalytics] Remaining Extractive Text Summarization implementation work

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/azure/azure-sdk-for-net/issues/22773.

Input action already implemented. This PR adds all output models and corresponding transforms for the Extractive Test Summarization API.

There are two major issues to solve after this one:
- Update the model factory for mocking the new models. Will do it in a separate PR to avoid polluting this one. Tracked by https://github.com/Azure/azure-sdk-for-net/issues/22836.
- Add live tests for the new API. Currently we only have a few mock tests that don't cover the whole usage scenarios. We still haven't updated our endpoint to version 3.2-preview.1, and we'll have a separate PR for that. Live testing work tracked by https://github.com/Azure/azure-sdk-for-net/issues/22839.

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Generated/Models/AnalyzeTasks.Serialization.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ internal static AnalyzeTasks DeserializeAnalyzeTasks(JsonElement element)
             Optional<IReadOnlyList<KeyPhraseExtractionTasksItem>> keyPhraseExtractionTasks = default;
             Optional<IReadOnlyList<EntityLinkingTasksItem>> entityLinkingTasks = default;
             Optional<IReadOnlyList<SentimentAnalysisTasksItem>> sentimentAnalysisTasks = default;
-            Optional<IReadOnlyList<TasksStateTasksExtractiveSummarizationTasksItem>> extractiveSummarizationTasks = default;
+            Optional<IReadOnlyList<ExtractiveSummarizationTasksItem>> extractiveSummarizationTasks = default;
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Generated/Models/AnalyzeTasks.Serialization.cs:27]: Renamed `TasksStateTasksExtractiveSummarizationTasksItem` to `ExtractiveSummarizationTasksItem` internally. It makes no difference in the public API, but other `<...>TaskItem` types were renamed as well, so I'm just following the pattern to keep them organized.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs [hunk 2]
```diff
@@ -486,7 +519,7 @@ private static string[] parseActionErrorTarget(string targetReference)
             {
                 throw new InvalidOperationException("Expected an error with a target field referencing an action but did not get one");
             }
-            Regex _targetRegex = new Regex("#/tasks/(keyPhraseExtractionTasks|entityRecognitionPiiTasks|entityRecognitionTasks|entityLinkingTasks|sentimentAnalysisTasks)/(\\d+)", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
+            Regex _targetRegex = new Regex("#/tasks/(keyPhraseExtractionTasks|entityRecognitionPiiTasks|entityRecognitionTasks|entityLinkingTasks|sentimentAnalysisTasks|extractiveSummarizationTasks)/(\\d+)", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:522]: I would have totally missed this change if we didn't have the `AnalyzeOperationWithActionsError` test.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs [hunk 3]
```diff
@@ -651,8 +690,6 @@ private static IReadOnlyCollection<RecognizeEntitiesActionResult> ConvertToRecog
             {
                 tasksErrors.TryGetValue(index, out TextAnalyticsErrorInternal taskError);
 
-                tasksErrors.TryGetValue(index, out taskError);
-
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:655]: Removed redundant line. Not related to the new API.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs [hunk 4]
```diff
@@ -318,6 +318,39 @@ private static int ParseHealthcareEntityIndex(string reference)
 
         #endregion
 
+        #region Extract Summary
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:321]: Adding lots of transforms in this class to convert from the internal models used by CodeGen to the public models exposed to users.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationMockTests.cs [hunk 5]
```diff
@@ -810,6 +810,11 @@ public async Task AnalyzeOperationWithActionsError()
                         ""code"": ""InvalidRequest"",
                         ""message"": ""Some error"",
                         ""target"": ""#/tasks/sentimentAnalysisTasks/0""
+                      },
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeOperationMockTests.cs:813]: Since we're parsing the service response now, updated one more non-live test we had to include the new API (tests the behavior in case of failures).

Live tests will be added in a following PR.

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,7 @@
 ## 5.2.0-beta.1 (Unreleased)
 
 ### Features Added
+- Added property `ExtractSummaryActions` to `TextAnalyticsActions` to support the new 'extractive text summarization' API. This action can be used to extract relevant sentences from an input document.
```
Comment (deyaaeldeen) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:6]: I think we need to be careful about describing this action because it applies a well-known NLP analysis and "relevant sentences" does not sound like the best description. Here is one suggestion:
```suggestion
- Added property `ExtractSummaryActions` to `TextAnalyticsActions` to support the new 'extractive text summarization' API. This action can be used to get a summary for the input document by extracting the most relevant sentences.
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs [hunk 7]
```diff
@@ -610,7 +610,8 @@ internal static LinkedEntityMatch LinkedEntityMatch(double confidenceScore = def
                 recognizeEntitiesActionResults.ToList(),
                 recognizePiiEntitiesActionResults.ToList(),
                 recognizeLinkedEntitiesActionsResults.ToList(),
-                analyzeSentimentActionsResults.ToList());
+                analyzeSentimentActionsResults.ToList(),
+                null); // TODO: include extractive summarization API (https://github.com/Azure/azure-sdk-for-net/issues/22836).
```
Comment (deyaaeldeen) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs:614]: I understand you want to do it in another PR but I am not sure why not doing it here?

**Review by deyaaeldeen — APPROVED** [review]
Thanks a lot for making the life of the reviewers much easier by breaking the work into multiple clean PRs and providing extensive guidance in the comments about what to look at. Much appreciated! I left two minor comments and I would highly appreciate taking action on the changelog one. Keep up the great work!

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,7 @@
 ## 5.2.0-beta.1 (Unreleased)
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:4]: I guess we missed it in the previous PR, we need to add the service bump version

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs (hunk 7 above)
Comment (kinelski) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs:614]: Now that you mention it, all changes would happen in the `TextAnalyticsModelFactory` class, so it probably wouldn't be a big deal. Just wanted to avoid adding extra changes to this PR (it's already larger than I was expecting). Also, the factory is a .NET-specific type as far as I know. I can update it in a separate PR and tag only Mariana so folks from other languages don't need to worry about it.

Comment (deyaaeldeen) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs:614]: Sounds good.

**Review by deyaaeldeen — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:522]: I actually added the test because when the service added sentiment analysis task, I missed it. 
Glad before GA we added the test and that it is helpful :)

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/SummarySentence.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+        /// <summary>
+        /// Gets the sentence text as it appears in the input document.
+        /// </summary>
+        public string Text { get; }
+
+        /// <summary>
+        /// Gets the rank score of the sentence, measuring how relevant the sentence is
+        /// to the input document. The value is set between [0.0, 1.0], with higher
+        /// values indicating a higher relevance.
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/SummarySentence.cs:31]: Do we know for sure that the rank score is between 0.0-1.0? I remember seeing examples showing negative numbers, but maybe that has since been fixed. @mssfang do you know?

**Review by maririos — APPROVED** [review]
Could you share an API view with the changes?

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Generated/Models/AnalyzeTasks.Serialization.cs (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/Generated/Models/AnalyzeTasks.Serialization.cs:27]: just to easy the use internally. it was getting hard to read those crazy types :P

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md (hunk 8 above)
Comment (kinelski) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:4]: Updated.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/SummarySentence.cs (hunk 9 above)
Comment (kinelski) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/SummarySentence.cs:31]: If I'm not mistaken, the range was mentioned in one of our design meetings, but it's not documented anywhere. We may need to double check with the service. Created https://github.com/Azure/azure-sdk-for-net/issues/22871 for tracking.

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md (hunk 8 above)
Comment (kinelski) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:4]: Noticed my push didn't work and the changelog was not updated. Adding the change to the next PR.
